File: metaData/EMC_str2double.m
function [ output_double ] = EMC_str2double( input_str )
    %Make a local copy of the default cluster, and modify the job storage
    %location
    
      output_double = str2double(input_str);
      if (isnan(output_double))
        output_double = str2num(input_str);
        isempty(output_double)
        if (isempty(output_double))   
        
          err_msg = sprintf('EMC_str2double: input string is not a number!\nReceived: %s', input_str);
          error(err_msg);
        end
      end
end

File: metaData/EMC_str2double.py
from numpy import ones

def EMC_str2double(input_str):
    try:
        output_double = float(input_str)
    except ValueError:
        try:
            output_double = eval(input_str)
            if output_double is None:
                raise ValueError
        except ValueError:
            try:
                # check for ones() is in the string and use the numpy function to replace it
                if 'ones(' in input_str:
                    input_str = input_str.replace('ones(', 'ones((')
                    input_str = input_str.replace(')', ', dtype=float)')
                    print(input_str)
            except ValueError:
                err_msg = f"EMC_str2double: input string is not a number!\nReceived: {input_str}"
                raise ValueError(err_msg)
    return output_double